src/lang/ir.cc

```diff
@@ -132,6 +298,802 @@ Expr Any::make() {
   return Expr(n);
 }
 
+Stmt LetStmt::make(Var var, Expr value, Stmt body) {
+  CHECK(value.defined());
+  CHECK(body.defined());
+  CHECK_EQ(value.type(), var.type());
+
+  NodePtr<LetStmt> node = make_node<LetStmt>();
+  node->var = std::move(var);
+  node->value = std::move(value);
+  node->body = std::move(body);
+  return Stmt(node);
+}
+
+Stmt AttrStmt::make(NodeRef node,
+                    std::string attr_key,
+                    Expr value,
+                    Stmt body) {
+  auto n = make_node<AttrStmt>();
+  n->node = node;
+  n->attr_key = std::move(attr_key);
+  n->value = std::move(value);
+  n->body = std::move(body);
+  return Stmt(n);
+}
+
+Stmt AssertStmt::make(Expr condition, Expr message, Stmt body) {
+  CHECK(condition.defined());
+  CHECK(message.type() == Int(32) ||
+        message.as<StringImm>())
+      << "TypeError: AssertStmt message must be an int or string:"
+      << message << "\n";
+
+  NodePtr<AssertStmt> node = make_node<AssertStmt>();
+  node->condition = std::move(condition);
+  node->message = std::move(message);
+  node->body = std::move(body);
+  return Stmt(node);
+}
+
+Stmt ProducerConsumer::make(FunctionRef func, bool is_producer, Stmt body) {
+  CHECK(body.defined());
```

Comment:
#define CHECK_DEFINE(x) CHECK(x.defined()) << #x << " is not defined";

Reply:
The error message of CHECK will automatically display the content, which is self-explained

Reply:
it will be CHECK: body.defined() failed